src/PostImage/PostImage.js

```diff
@@ -1,14 +1,24 @@
 import styles from './PostImage.module.css';
+import cx from 'classnames';
 
-export function PostImage({ imageURL, fraction, aspectRatio }) {
+export function PostImage({
+  imageURL,
+  fraction,
+  aspectRatio,
+  isLoading,
+  setIsLoading,
+}) {
   return (
     <img
-      className={styles.postImg}
+      className={cx(styles.postImg, {
+        [styles.isLoading]: isLoading,
+      })}
       src={imageURL}
       alt="post"
       loading="lazy"
       width={`${fraction * 100}%`}
       style={{ aspectRatio }}
+      onLoad={() => setIsLoading(false)}
```

Comment:
I think this will be throwing errors for items that are not the first. You need to check if the function is defined before calling it.
```suggestion
      onLoad={() => setIsLoading?.(false)}
```

Reply:
Done